cli/azd/cmd/container.go

```diff
@@ -549,6 +549,15 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		})
 	})
 
+	container.MustRegisterSingleton(func(lazyConfig *lazy.Lazy[*project.ProjectConfig]) (project.IsProjectADE, error) {
```

Comment:
We should not really need to register things like this.

Reply:
It's a beautiful way to break dependencies between components by using the ioc.
The ioc knows everything and has access to all.
That's a real benefit which we should use more, instead of using the ioc to couple components together.

Think about it..  It meas that if a package, like `project` needs to inspect or use a type from another package like `platform`, it defines its own type to pull what it needs from the `platform` and ask the ioc to be the interface to pull the information from the `platform`.  The `platform` package just exposes a function, like an api, to interact with other packages, instead of doing coupling.